src/location.js

```diff
@@ -10,21 +10,45 @@ const Location = ({ homeUrl, children }) => {
   // Set the location state to a parsed url and a compiled url.
   const [parsedUrl, setParsedUrl] = useState(parseJsonApiUrl(homeUrl));
   const [locationUrl, setLocationUrl] = useState(compileJsonApiUrl(parsedUrl));
-  const [document, setDocument] = useState(null);
+  const [responseDocument, setDocument] = useState(null);
+
+  const setUrl = newLocationUrl => {
```

Comment:
Should we reserve `set` methods for the setState hooks?

Reply:
I thought about that but chose to use "set" because it's kind of decorating/combining `setParsedUrl` and `setLocationUrl` and also it is part of the exposed API of the context along with `setSort`, `setFragment` etc.

All that said, if you have a strong opinion about it, I will gladly change it :)

Reply:
not a strong opinion, just a convention that I wasn't sure if we should follow. I'm fine with leaving it I just had a moment of confusion trying to see where it came from.